### PR TITLE
[Admin ] - Gab/auth integration

### PR DESCRIPTION
**Pull request by gidoo5331**
I get this error when i run yarn lint
(./lib/cookies.config.ts
8:14  Error: 'error' is defined but never used.  @typescript-eslint/no-unused-vars
16:14  Error: 'error' is defined but never used.  @typescript-eslint/no-unused-vars
)
should I just use a console.log or is there any other way to solve it?

Screenshot from 2024-11-06 09-24-13

Screenshot from 2024-11-06 09-26-03

### CODE REVIEW

File: apps/admin/lib/cookies.config.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import Cookies from 'js-cookie'
+
+class Auth {
+  getCipher(key: string) {
+    try {
+      const token = Cookies.get(key)
+      return token != undefined ? JSON.parse(token) : null
+    } catch (error) {}
```
Comment (Bendomey) [inline apps/admin/lib/cookies.config.ts:8]:
```suggestion
    } catch () {}
```

File: apps/admin/lib/cookies.config.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  }
+
+  setCipher<T>(key: string, data: T) {
+    try {
+      Cookies.set(key, JSON.stringify(data), {
+        expires: 1, // 1 day
+      })
+    } catch (error) {}
```
Comment (Bendomey) [inline apps/admin/lib/cookies.config.ts:16]:
```suggestion
    } catch () {}
```

**Review by Bendomey — CHANGES_REQUESTED** [review]

File: apps/admin/lib/cookies.config.ts (hunk 2 above)
Comment (gidoo5331) [reply apps/admin/lib/cookies.config.ts:16]: Screenshot from 2024-11-06 09-52-34
Screenshot from 2024-11-06 09-53-09

File: apps/admin/providers/auth.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+              }
+            },
+            onError: e => {
+              toast({
+                title: "Session Expired",
+                variant: "destructive",
+                description: 'Login to access resource',
+                duration: 5000,
+              });
+              Router.replace(`/login`)
```
Comment (Bendomey) [inline apps/admin/providers/auth.tsx:55]: probably need the return_to thing here

```suggestion
              Router.replace(`/login?redirect_to=${pathname}`)
```

**Review by Bendomey — CHANGES_REQUESTED** [review]

File: apps/admin/lib/cookies.config.ts (hunk 2 above)
Comment (Bendomey) [reply apps/admin/lib/cookies.config.ts:16]: better to remove the try/catch block

**Review by Bendomey — APPROVED** [review]

**Review by arkhurst — COMMENTED** [review]
👍🏽

**Pull request merged**